brave/brave-core
# [Android] reorganises Settings and Privacy settings, sets shields global prefs

Resolves https://github.com/brave/brave-browser/issues/7487

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

Please check the latest figma file included in the issue for the UI/UX changes:
- all settings found in figma should appear in the exact order in the app
- all icons should correspond to the right setting
- any other inconsistencies

Probably all (or most) settings will need to be re-checked just to be sure.
Please look for any missing settings  or things that might not make sense.

The way the settings bellow work has been changed :

Search engines
- opens a new preference screen where you can choose the default search engine for private and standard tabs
- the behaviour after choosing should be as before and reflect in the app's behaviour

Testing for the new Brave shields & privacy settings -> shields global defaults:

 - Block ads & trackers - set to aggressive 
  1. On duckduckgo search for something like "samsung galaxy s10"
  2. Check the result of the search: 
   -  there should be no hits that contain Ads
 - Block ads & trackers - set to standard or allow
  1. Open a new tab and repeat the steps above. The result should contain a few Ads hits in the beginning of the result list

-Block cookies
  - block all cookies:
    - Open a new tab and try to log on facebook. you shouldn't be able log in
  - block cross-site cookies and allow should allow you to login

 -Fingerprinting -test plan to be determined

- Block scripts -on should show a blank page on Facebook

Testing for Brave shields & privacy settings -> Clear browsing data - Clear data on exit

1.When it's on, after the app is closed (swiped out) re-opening it should result in the following data to be wiped (it wipes it for the whole history):
- browsing history
- cookies
- passwords
- forms
- site settings
- cache
It does not closes open tabs (for that there is the separate close all tabs on exit setting that can be used in combination)

Testing for  Brave shields & privacy settings -> Other  - WebRTC
- After entering the preference screen for WebRTC there should be a Learn More link that should go to https://support.brave.com/hc/en-us/articles/360017989132-How-do-I-change-my-Privacy-Settings-#webrtc
- The browser functions and behaviour should be the same
